zstackctl/zstackctl/ctl.py

```
@@ -8979,6 +9000,16 @@ def chown_to_zstack():
                 info('change permission to user zstack')
                 shell('chown -R zstack:zstack %s' % os.path.join(ctl.zstack_home, '../../'))
 
+            if not args.skip_precheck and not need_download and new_war.path:
+                ret = subprocess.run(
```

Ruff (0.15.7)

[error] `subprocess` call: check for execution of untrusted input

(S603)